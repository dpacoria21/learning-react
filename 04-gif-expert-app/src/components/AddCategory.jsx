import { useState } from "react"
import  PropTypes  from 'prop-types'

export const AddCategory = ({onNewCategory}) => {


    let [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => setInputValue(target.value);

    const onSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length <= 1) return;

        onNewCategory(inputValue);

        setInputValue('');

    }

    return (

        <form onSubmit={onSubmit}>
            <input 
                type="text" 
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>

    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}