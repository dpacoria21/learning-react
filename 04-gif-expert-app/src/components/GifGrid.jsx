import { getGifs } from "../helpers/getGifs";
import { useState, useEffect } from "react";


export const GifGrid = ({category}) => {   


    const [counter, setCounter] = useState(10);
    useEffect(() => {
        getGifs(category);
    }, []);

    // console.log(counter);

    return (
        <>
            <h3>{category}</h3>        
            <p>Hola Mundo</p>
            <button onClick={() => setCounter(counter+1)}>+1</button>
        </>
    )
}