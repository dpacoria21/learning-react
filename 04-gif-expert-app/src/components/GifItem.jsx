import PropTypes from 'prop-types';

export const GifItem = ({title, url}) => {
    return (
        <div className="card">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
        // <li>{title}</li>

    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

