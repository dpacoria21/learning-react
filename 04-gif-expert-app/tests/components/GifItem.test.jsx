import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem"

describe('GifItem testing', () => {
    const title = 'One Punch';
    const url = 'https://one-punch.com/saitama.jpg';
    test('Debe de regresar un arreglo de valores', () => {
        const {container} = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={title} url={url} />);
        // screen.debug();
        // console.log(screen.getByRole('img').alt);
        // expect( screen.getByRole('img').src ).toBe(url);
        // expect( screen.getByRole('img').alt ).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe mostrar el titulo en el componente', () => {
        render(<GifItem title={title} url={url} />);
        // screen.debug();
        const testTitle = screen.getByText(title);
        // console.log(testTitle);
        expect(testTitle).toBeTruthy();

    })
});