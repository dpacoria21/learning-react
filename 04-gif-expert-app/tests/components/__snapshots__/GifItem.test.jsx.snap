// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem testing Debe de regresar un arreglo de valores 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
